pkg/controller/perconaxtradbcluster/perconaxtradbcluster_controller.go

```diff
@@ -333,6 +339,60 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 	return nil
 }
 
+func (r *ReconcilePerconaXtraDBCluster) checkSecret(cr *api.PerconaXtraDBCluster, namespace string) error {
+	if cr.Spec.PXC.AllowUnsafeConfig == true {
+		return nil
+	}
+	secretObj := corev1.Secret{}
+	err := r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: namespace,
+			Name:      cr.Spec.PXC.SSLSecretName,
+		},
+		&secretObj)
+	if err != nil && errors.IsNotFound(err) {
+		var (
+			issuerKind = "ClusterIssuer"
+			issuerName = cr.Spec.PXC.SSLSecretName + "-ca"
+		)
+		issuer := cm.ClusterIssuer{}
+		issuer.Namespace = namespace
+		issuer.Kind = issuerKind
+		issuer.Name = issuerName
+		issuer.Spec.SelfSigned = &cm.SelfSignedIssuer{}
+		err := r.client.Create(context.TODO(), &issuer)
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return err
+		}
+		issuer = cm.ClusterIssuer{}
+		err = r.client.Get(context.TODO(),
+			types.NamespacedName{
+				Name: issuerName,
```

Comment:
not sure if it's legit to omit "Namespace", see: https://github.com/kubernetes/apimachinery/blob/master/pkg/types/namespacedname.go#L23